vector/src/main/java/im/vector/app/features/settings/FontScalePreferences.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.settings
+
+import android.content.SharedPreferences
+import androidx.annotation.StringRes
+import androidx.core.content.edit
+import im.vector.app.R
+import im.vector.app.core.di.DefaultPreferences
+import im.vector.app.core.utils.SystemSettingsProvider
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Stores and returns font scale settings using shared preferences.
+ */
+interface FontScalePreferences {
+    /** Defines whether to use system settings for font scale or not.
+     * @param useSystem true to use system settings, false to use app settings
+     */
+    fun setUseSystemScale(useSystem: Boolean)
+
+    /** Returns whether to use system settings for font scale or not.
+     * @return useSystem true to use system settings, false to use app settings
+     */
+    fun getUseSystemScale(): Boolean
+
+    /** Returns font scale taking in account [useSystemScale] setting.
+     * @return App setting for font scale if [getUseSystemScale] returns false, system setting otherwise
+     */
+    fun getResolvedFontScaleValue(): FontScaleValue
+
+    /** Returns persisted app font scale.
+     * @return app setting for font scale
+     */
+    fun getAppFontScaleValue(): FontScaleValue
+
+    /** Sets and stores app font scale setting value.
+     * @param fontScaleValue value to be set and saved
+     */
+    fun setFontScaleValue(fontScaleValue: FontScaleValue)
+
+    /** Returns list of all available font scale values.
+     * @return list of values
+     */
+    fun getAvailableScales(): List<FontScaleValue>
+}
+
+/**
+ * Object to manage the Font Scale choice of the user.
+ */
+@Singleton
```

Comment:
what's the reason for using `@Singleton`? I would lean against the usage as the class isn't heavy to instantiate and doesn't directly hold any mutable state

once the classloader encounters a singleton it'll be in memory for the lifetime of the application (this class wouldn't be needed once the app is in the background for example)